experimental/Origami/src/Origami.jl

```diff
@@ -0,0 +1,333 @@
+            if i^(sigma * x^-1 * y^-1) == i^(y*x*(x*y)^-1)
+                b = b + 1
+            end
+        end
+
+        # TODO can n_inv yields floating point errors?
+        if n_inv * (g - 1 - b * 0.5) + 1 == 0
+            return true
+        end
+    end
+
+    return false
+end
+
+function cylinder_structure(o::Origami)
+    gap_obj = GAP.Globals.CylinderStructure(GapObj(o))
+    cyl_lists = Vector{Vector{Int}}(gap_obj)
+    # TODO do we want the structure as a list of length-two lists or as
+    # a list of tuples? Current solution uses tuples.
+    cyl_tuples = [(l[1], l[2]) for l in cyl_lists]
+    return cyl_tuples
+end
+
+function veech_group_and_orbit(o::Origami)
+    return GAP.Globals.VeechGroupAndOrbit(GapObj(o))
+end
+
+function veech_group_is_even(o::Origami)
+    return GAP.Globals.VeechGroupIsEven(GapObj(o))::Bool
+end
+
+function are_equivalent(o1::Origami, o2::Origami)
+    return GAP.Globals.OrigamisEquivalent(GapObj(o1), GapObj(o2))::Bool
+end
+
+# TODO why is this not in canonical.jl?
+function normalform_conjugators(o::Origami)
+    x = horizontal_perm(o)
+    y = vertical_perm(o)
+    n = degree(o)
+    G = []
+
+    # Starting from each of the vertices found above, do a breadth-first search
+	# and list the vertices in the order they appear.
+	# This defines a permutation l with which we conjugate x and y.
+	# From the resulting list of pairs of permutations (all of which are by
+	# definition simultaneously conjugated to (x,y)) we choose the
+	# lexicographically smallest one as the canonical form.
+    for i in 1:n
+        L = fill(0, n)
+        seen = fill(false, n)
+        Q = [i]
+        seen[i] = true
+        numSeen = 1
+        L[i] = 1
+        while numSeen < n
+            v = popfirst!(Q)
+            wx = v^x
+            wy = v^y
+            if !seen[wx]
+                push!(Q, wx)
+                seen[wx] = true
+                numSeen = numSeen + 1
+                L[wx] = numSeen
+            end
+            if !seen[wy]
+                push!(Q, wy)
+                seen[wy] = true
+                numSeen = numSeen + 1
+                L[wy] = numSeen
+            end
+        end
+        push!(G, L)
+    end
+
+    return perm.(G)
+end
+
+function point_reflections(o::Origami)
+    if !veech_group_is_even(o)
+        throw("VeechGroup must contain -1")
+    end
+
+    h = horizontal_perm(o)
+    v = vertical_perm(o)
+    o1 = origami(h^-1, v^-1)
+    h1 = horizontal_perm(o1)
+    v1 = vertical_perm(o1)
+    G = normalform_conjugators(o)
+    G1 = normalform_conjugators(o1)
+
+    f(i) = origami(i^-1 * h * i, i^-1 * v * i)
+    f1(i) = origami(i^-1 * h1 * i, i^-1 * v1 * i)
+
+    # origamis derived from the permutations above
+    O = f.(G)
+    # we need to calculate these to test find k s.t.
+    # sigma_i *origami *sigma_i^-1=delta_k(i)*origami_1*delta_k(i)^-1
+    O1 = f1.(G1)
+
+    # fitting the permuations together
+    result::Vector{PermGroupElem} = []
```

Comment:
The original code means: "create an empty `Vector{Any}` then convert it into a `Vector{PermGroupElem}`. Better to directly create an empty `Vector{PermGroupElem}`:
```suggestion
    result = PermGroupElem[]
```